pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -203,6 +234,40 @@ func (p *staticPolicy) RemoveContainer(s state.State, containerID string) error
 	return nil
 }
 
+func (p *staticPolicy) inheritCPUs(s state.State, pod *v1.Pod, numCPUs int) (cpuset.CPUSet, error) {
+	// Get a sorted slice of all init Container cpuSets.
+	var cpuSets []*cpuset.CPUSet
+	for _, container := range pod.Spec.InitContainers {
+		containerID, err := p.containerMap.Get(pod, &container)
+		if err != nil {
+			continue
+		}
+		if cset, ok := s.GetCPUSet(containerID); ok {
+			cpuSets = append(cpuSets, &cset)
+		}
+	}
+	sort.Slice(cpuSets, func(i, j int) bool {
+		return cpuSets[i].Size() < cpuSets[j].Size()
+	})
+
+	// Inherit up to numCPUs from the largest CPUSet in cpuSets.
+	if len(cpuSets) == 0 {
+		return cpuset.NewCPUSet(), nil
+	}
+
+	if cpuSets[0].Size() < numCPUs {
+		numCPUs = cpuSets[0].Size()
+	}
+
+	result, err := takeByTopology(p.topology, *cpuSets[0], numCPUs)
+	if err != nil {
+		return cpuset.NewCPUSet(), err
```

Comment:
Should we log something here to indicate that `takeByTopology` failed?

Reply:
That happens at the call site